GoogleForCreators/web-stories-wp
# Dashboard:  Scroll to Top (my stories + storybook)

# Overview 
Part of the new UI that Sam made for the dashboard is the ability to scroll to the top of a given view once you scroll down at all. 

## Implementation 
- We're tracking the current Y axis with a ref hook and debouncing a callback attached to a window event listener of scroll to find out when the y axis isn't 0.  I believe using a window event listener + the debounced callback here straight forward and comes with the least amount of complexity. Since scroll to top is it's own thing outside of any kind of scrolling I didn't want to tie its functionality to anything outside of it. 
- So, this doesn't work unless the body.web-story_page_stories-dashboard #web-stories-dashboard doesn't set the height for the view! I know that @carlos-kelly added the height calc to respect the admin toolbar height here a few days ago. I moved that height calc to the side nav and I _think_ everything is working fine? I can't find any other way to get `scrollTo` to work with this. Apparently it's a wordpress peculiarity - slack overflow-ers from 5 years ago guided me to my answer, was trying a bunch of other things. 

## Testing 
- See in storybook: Dashboard/Components/ScrollToTop 
 - See in dashboard on my stories view 
 - In either spot, just scroll and then you'll see the button show up. When you scroll back to the top (with the button or with your mouse/trackpad) the button should disappear. 

## Misc updates 
- renamed two svg files to use camelcasing since we're slowly moving to that consistency. 

## To Do
- I don't have the right SVG for this component, I just used the dropdown arrow as a placeholder. Once Sam posts the designs in figma I'll be able to import the SVG. (made this issue: https://github.com/google/web-stories-wp/issues/1321) 

I based my changes off of this video: https://drive.google.com/drive/u/0/folders/1onGbNRFVAalJXtbYYFigSbdG-nHUE7As 

**(here's a gif version)**
![scroll to top](https://user-images.githubusercontent.com/10720454/80133732-8695d400-8552-11ea-8b48-5187e18cf974.gif)

**this branch**
![dashboard scroll to top](https://user-images.githubusercontent.com/10720454/80136007-f5c0f780-8555-11ea-9165-aa9542b501b1.gif)
